src/pages/tracked-contracts/TrackedContractsList.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import { Card, CardBody, CardHeader } from '../../components/common';
+import SearchInput from '../../components/common/Input';
+import mockDataChain from './mock-data';
+
+type ChainType = 'evm' | 'substrate';
+
+interface Chain {
+  name: string;
+  type: ChainType;
+  project: string;
+  url: string;
+  credentials: {
+    // TODO: Ask potentially BE to have structure camelCase in case or we can just disable the warning in eslint
+    /* eslint-disable camelcase */
+    private_key?: string;
+    public_key?: string;
+  };
+  tracked_contracts: string[];
+}
+
+const TrackedContractsList: React.FC<{}> = () => {
+  const [chains, setChains] = useState<Chain[]>([]);
+  const [activeChain, setActiveChain] = useState<Chain>();
+  const [searchChain, setSearchChain] = useState('');
+
+  useEffect(() => {
+    setChains(mockDataChain as Chain[]);
+  }, []);
+
+  const allChains = chains
+    .filter((data) => {
+      if (!searchChain) {
+        return data;
+      }
+      if (data.name.includes(searchChain.trim().toLocaleLowerCase())) {
+        return data;
+      }
+      return null;
+    })
+    .map((chain) => {
+      return (
+        <div
+          className={`focus:outline-none py-4 capitalize cursor-pointer ${
+            chain.name === activeChain?.name ? 'font-bold text-primary' : ''
+          }`}
+          role="button"
+          onClick={() => setActiveChain(chain)}
+          onKeyPress={() => setActiveChain(chain)}
+          tabIndex={0}
+        >
+          {chain.name}
+        </div>
+      );
+    });
```

Comment:
A bit nicer:
`
 const allChains = chains
    .filter((chain) => !searchChain)
    .filter((chain) => chain.name.includes(searchChain.trim().toLocaleLowerCase()))
    .map((chain) => (
        <div
          className={`focus:outline-none py-4 capitalize cursor-pointer ${
            chain.name === activeChain?.name ? 'font-bold text-primary' : ''
          }`}
          role="button"
          onClick={() => setActiveChain(chain)}
          onKeyPress={() => setActiveChain(chain)}
          tabIndex={0}
        >
          {chain.name}
        </div>
      ));
`

Reply:
Also, I would try to move out the div view and present it as a component

Reply:
The filter like that wouldn't actually work but yeah simplifying it and moving it to a separate component.